packages/server/src/api/controllers/row/ExternalRequest.ts

```diff
@@ -697,24 +734,46 @@ export class ExternalRequest<T extends Operation> {
         continue
       }
       for (let row of rows) {
-        const filters = buildFilters(generateIdForRow(row, table), {}, table)
-        // safety check, if there are no filters on deletion bad things happen
-        if (Object.keys(filters).length !== 0) {
-          const op = isMany ? Operation.DELETE : Operation.UPDATE
-          const body = isMany ? null : { [colName]: null }
-          promises.push(
-            getDatasourceAndQuery({
-              endpoint: getEndpoint(tableId, op),
-              body,
-              filters,
-            })
-          )
+        const promise = removeRelationships(
+          generateIdForRow(row, table),
+          table,
+          isMany,
+          colName
+        )
+        if (promise) {
+          promises.push(promise)
         }
       }
     }
     await Promise.all(promises)
   }
 
+  async removeRelationshipsToRow(table: Table, rowId: string) {
+    const row = await this.getRow(table, rowId)
+    const related = await this.lookupRelations(table._id!, row)
+    for (let column of Object.values(table.schema)) {
+      if (
```

Comment:
I created a little suite of type-guard functions for this purpose: https://github.com/Budibase/budibase/blob/c60e2bf41cd09483d86000cc7db8d4ccf46da92f/packages/types/src/documents/app/table/schema.ts#L183-L187.

You could, e.g., write:

```javascript
if (!isManyToOne(column)) {
    continue
}
```

And the type-guard on `isManyToOne` would obviate the need for the cast on line 761.